docs/01-get-started/04-working-with-the-database.md

```diff
@@ -0,0 +1,338 @@
+
+```bash
+$ cd magic_recipe/magic_recipe_server
+$ serverpod create-migration
+```
+
+You will notice that there will be a new entry in your "migrations" folder - serverpod creates these migrations "step by step" - each time you have changes which are relevant to the database and run `serverpod create-migrations` a new migration file will be created. This is a good way to keep track of the changes you make to the database and to be able to roll back changes if needed.
+
+## Writing to the database
+
+Let's save all created recipes to the database. First we want to automatically store new recipes in the database. Because the `Recipe` now has a `table` it is now not just a serializable entity, but also a `TableRow` - that means that we can use the `insertRow` method to store it in the database.
+
+```dart
+// recipe_endpoint.dart
+
+
+class RecipeEndpoint extends Endpoint {
+  Future<Recipe> generateRecipe(Session session, String ingredients) async {
+
+  // ...
+
+    final recipe = Recipe(
+      author: 'Gemini',
+      text: responseText,
+      date: DateTime.now(),
+      ingredients: ingredients,
+    );
+
+    // --- Add this ---
+
+    // Save the recipe to the database, but don't block the response
+    unawaited(Recipe.db.insertRow(session, recipe));
+
+    // --- End of added code ---
+
+    return recipe;
+  }
+
+```
+
+## Reading from the database
+
+Next we want to add a new endpoint that will return all the favourite recipes from the database. We will create a new method in the `RecipeEndpoint` class that will return all the recipes from the database.
+
+To make sure that we get them in the correct order, we can sort them by the date they were created.
+
+```dart
+// recipe_endpoint.dart
+
+
+class RecipeEndpoint extends Endpoint {
+
+  // ...
+
+  /// This method is used to get all the generated recipes from the database.
+  Future<List<Recipe>> getRecipes(Session session) async {
+    // Get all the recipes from the database, sorted by date
+    return Recipe.db.find(
+      session,
+      orderBy: (t) => t.date,
+      orderDescending: true,
+    );
+  }
+```
+
+:::info
+The when adding a `table` to the model class definition, the model will now give you access to the database, specifically to the `recipes` table through `Recipe.db` (e.g. `Recipe.db.find(session)`.
+The `insertRow` method is used to insert a new row in the database. The `find` method is used to query the database and get all the rows of a specific type. See [CRUD](../concepts/database/crud) and [relation queries](../concepts/database/relation-queries) for more information.
+:::
+
+## Generate the code
+
+Like before, when you change something that has an effect on the client code, you need to run `serverpod generate` - we don't need to run `serverpod create-migrations` again because we already created the migrations in the previous step.
```

Comment:
```suggestion
Like before, when you change something that has an effect on the client code, you need to run `serverpod generate`. We don't need to run `serverpod create-migrations` again because we already created the migrations in the previous step.
```